File: Cpp/1749_Maximum_Absolute_Sum_of_Any_Subarray.cpp
class Solution {
public:
    int maxAbsoluteSum(vector<int>& nums) {
        int max_sum = 0;
        int min_sum = INT_MAX;
        int positive = 0;
        int negative = 0;
        
        for(auto& num : nums) {
            positive = max(positive + num, 0);
            max_sum = max(max_sum, positive);
            
            negative = min(negative + num, 0);
            min_sum = min(min_sum, negative);
        }
        
        return max(max_sum, -min_sum);
    }
};

File: Kotlin/1749_Maximum_Absolute_Sum_of_Any_Subarray.kt
class Solution {
    fun maxAbsoluteSum(nums: IntArray): Int {
        var maxSum = 0
        var minSum = Integer.MAX_VALUE
        var positive = 0
        var negative = 0
        
        for(num in nums) {
            positive = maxOf(positive + num, 0)
            maxSum = maxOf(maxSum, positive)
            
            negative = minOf(negative + num, 0)
            minSum = minOf(minSum, negative)
        }
        
        return maxOf(maxSum, -minSum)
    }
}